bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/TextValue.java

```diff
@@ -37,30 +38,61 @@
 @NonNullByDefault
 public class TextValue extends Value {
     private final @Nullable Set<String> states;
+    private final @Nullable Set<String> commands;
 
     /**
-     * Create a string value with a limited number of allowed states.
+     * Create a string value with a limited number of allowed states and commands.
      *
      * @param states Allowed states. Empty states are filtered out. If the resulting set is empty, all string values
      *            will be allowed.
+     * @param commands Allowed commands. Empty commands are filtered out. If the resulting set is empty, all string
+     *            values
+     *            will be allowed.
      */
-    public TextValue(String[] states) {
+    public TextValue(String[] states, String[] commands) {
         super(CoreItemFactory.STRING, List.of(StringType.class));
         Set<String> s = Stream.of(states).filter(not(String::isBlank)).collect(Collectors.toSet());
         if (!s.isEmpty()) {
             this.states = s;
         } else {
             this.states = null;
         }
+        s = Stream.of(commands).filter(not(String::isBlank)).collect(Collectors.toSet());
+        if (!s.isEmpty()) {
+            this.commands = s;
+        } else {
+            this.commands = this.states;
```

Comment:
Not all states are always automatically allowed commands. If no commands are provided, why not simply leave the commands empty here?